crates/cheatcodes/spec/src/vm.rs

```diff
@@ -2888,6 +2888,28 @@ interface Vm {
     /// catch (bytes memory interceptedInitcode) { initcode = interceptedInitcode; }
     #[cheatcode(group = Utilities, safety = Unsafe)]
     function interceptInitcode() external;
+
+    /// Generates the hash of the canonical EIP-712 type representation.
+    ///
+    /// Supports 2 different inputs:
+    ///  1. Name of the type (i.e. "Transaction"):
+    ///     * requires previous binding generation with `forge bind-json`.
+    ///     * bindings will be retrieved from the path configured in `foundry.toml`.
+    ///
+    ///  2. String representation of the type (i.e. "Foo(Bar bar) Bar(uint256 baz)").
+    ///     * Note: the cheatcode will output the canonical type even if the input is malformated
+    ///             with the wrong order of elements or with extra whitespaces.
+    #[cheatcode(group = Utilities)]
```

Comment:
uh, sorry, just realized this - in forge-std we have checks for solc backward compatibility, so if you take this def and put in Vm.sol and run `forge build --skip test --use solc:0.6.2` it will fail with 
```
Error: Compiler run failed:
Error: Data location must be "calldata" for parameter in external function, but "memory" was given.
src/Vm.sol:2415:59: TypeError: Data location must be "calldata" for parameter in external function, but "memory" was given.
    function eip712HashType(string calldata bindingsPath, string memory typeName) external pure returns (bytes32 typeHash);
```

please change `typeNameOrDefinition` and `typeName` params to calldata instead memory